examples/pytorch/gin/train.py

```diff
@@ -0,0 +1,143 @@
+        return score_over_layer
+    
+def split_fold10(labels, fold_idx=0, seed=0, shuffle=True):
+    skf = StratifiedKFold(n_splits=10, shuffle=shuffle, random_state=seed)
+    idx_list = []
+    for idx in skf.split(np.zeros(len(labels)), labels):
+        idx_list.append(idx)
+    train_idx, valid_idx = idx_list[fold_idx]
+    return train_idx, valid_idx
+
+def evaluate(dataloader, device, model):
+    model.eval()
+    total = 0
+    total_correct = 0
+    for batched_graph, labels  in dataloader:
+        batched_graph = batched_graph.to(device)
+        labels = labels.to(device)
+        feat = batched_graph.ndata.pop('attr')
+        total += len(labels)
+        logits = model(batched_graph, feat)
+        _, predicted = torch.max(logits.data, 1)
+        total_correct += (predicted == labels.data).sum().item()
+    acc = 1.0 * total_correct / total
+    return acc
+
+def train(train_loader, val_loader, device, model):
+    # loss function, optimizer and scheduler
+    loss_fcn = nn.CrossEntropyLoss()
+    optimizer = optim.Adam(model.parameters(), lr=0.01)
+    scheduler = optim.lr_scheduler.StepLR(optimizer, step_size=50, gamma=0.5)
+    
+    # training loop    
+    for epoch in range(350):
+        model.train()
+        total_loss = 0
+        for batch, (batched_graph, labels) in enumerate(train_loader):
+            batched_graph = batched_graph.to(device)
+            labels = labels.to(device)
+            feat = batched_graph.ndata.pop('attr')
+            logits = model(batched_graph, feat)
+            loss = loss_fcn(logits, labels)
+            optimizer.zero_grad()
+            loss.backward()
+            optimizer.step()
+            total_loss += loss.item()
+        scheduler.step()
+        train_acc = evaluate(train_loader, device, model)
+        valid_acc = evaluate(val_loader, device, model)
+        print("Epoch {:05d} | Loss {:.4f} | Train Acc. {:.4f} | Validation Acc. {:.4f} "
+              . format(epoch, total_loss / (batch + 1), train_acc, valid_acc))
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--dataset', type=str, default="MUTAG",
+                        choices=['MUTAG', 'PTC', 'NCI1', 'PROTEINS'],
+                        help='name of dataset (default: MUTAG)')
+    args = parser.parse_args()
+    print(f'Training with DGL built-in GINConv module with a fixed epsilon = 0')
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+    
+    # load and split dataset
+    dataset = GINDataset(args.dataset, True, False) #  self_loop and disable one-hot encoding for input features
+    labels = [l for _, l in dataset]
+    train_idx, val_idx = split_fold10(labels)
+    
+    # create dataloader
+    train_loader = GraphDataLoader(dataset, sampler=SubsetRandomSampler(train_idx),
+                                   batch_size=128, pin_memory=torch.cuda.is_available())
```

Comment:
I think you need to specify `shuffle=True` for `train_loader`.

Reply:
Sure.

Reply:
I see. Thanks.